fix dead link in svelte typescript tutorial (#32108)
mdn/content: files/en-us/learn/tools_and_testing/client-side_javascript_frameworks/svelte_typescript/index.md

   export let todo: TodoType;
   ```

>    > **Note:** Another reminder: When importing a `.ts` file, you have to omit the extension. Check the [`modules-in-typescript` section](https://www.typescriptlang.org/docs/handbook/2/modules.html#modules-in-typescript) of the TypeScript manual for more information.

Author: ```suggestion
   > **Note:** Another reminder: When importing a `.ts` file, you could omit the extension. Check the [`Extensionless relative paths` section](https://www.typescriptlang.org/docs/handbook/modules/reference.html#extensionless-relative-paths) of the TypeScript manual for more information.
```

Author: I just did an experiment on my computer, finding that the `.ts` file could be imported with either `.js` extension or with no extention, but we can't import `.ts` file with `.ts` extension directly.

>    > **Note:** Another reminder: When importing a `.ts` file, you have to omit the extension. Check the [`modules-in-typescript` section](https://www.typescriptlang.org/docs/handbook/2/modules.html#modules-in-typescript) of the TypeScript manual for more information.

Author: ```suggestion
   > **Note:** Another reminder: When importing a `.ts` file, you should use the `.js` extension or no extension at all (no `.ts` extension in the import clause). Check the [`extensionless relative paths` section](https://www.typescriptlang.org/docs/handbook/modules/reference.html#extensionless-relative-paths) of the TypeScript manual for more information.
```

>    > **Note:** Another reminder: When importing a `.ts` file, you have to omit the extension. Check the [`modules-in-typescript` section](https://www.typescriptlang.org/docs/handbook/2/modules.html#modules-in-typescript) of the TypeScript manual for more information.

Author: ```suggestion
   > **Note:** Another reminder: When importing a `.ts` file, you will get an error if the `.ts` extension is included in the statement. Check the [`module resolution for bundlers, TypeScript runtimes, and Node.js loaders` section](https://www.typescriptlang.org/docs/handbook/modules/theory.html#module-resolution-for-bundlers-typescript-runtimes-and-nodejs-loaders) of the TypeScript manual for more information.
```

Author: "When importing a .ts file, you have to omit the extension." seems not to be wrong, but just mislead readers (readers may think that `.ts` files could only be imported without extension)

Reviewer: I think the #module-resolution-for-bundlers-typescript-runtimes-and-nodejs-loaders link is better because it points to the specific section where it talks about this restriction.
But the right place for this information is the first time that we mention you need to omit the extension. So I have done that further up in the doc. Here we don't need to repeat that detail, though it is still worth the reminder.

```suggestion
   Note that the `.ts` file extension is not allowed in the `import` statement, and has been omitted.
```

>    > **Note:** Another reminder: When importing a `.ts` file, you have to omit the extension. Check the [`modules-in-typescript` section](https://www.typescriptlang.org/docs/handbook/2/modules.html#modules-in-typescript) of the TypeScript manual for more information.

Author: ```suggestion
   > **Note:** Another reminder: When importing a `.ts` file, you will get an error if the `.ts` extension is included in the import statement. Check the [`modules theory` section](https://www.typescriptlang.org/docs/handbook/modules/theory.html) of the TypeScript manual for more information.
```
